### DISCUSSION TITLE
why book snapshot and delta have different data structures?

### DISCUSSION THREAD

**Discussion by lfnt3** [bmoscon/cryptofeed#767] [Q&A]
Hello,

@bmoscon 
Before I post my question let me express my excitement about this software project! It's an amazing package not only to be used with convenience but also as a source to learn from!

Now to my question.
The bid and ask value pairs of the orderbook datatype are provided as a simple dictionary structures for the snapshot (book).
But the delta bid and ask value pairs are provided as a lists of tuples.

Is there a reason why the delta is not provided as a dict like the snapshot?

Thanks!

**Comment by bmoscon** [comment]
none that i can think of
